new-components/reporters/slack/internal/reporter/reporter.go

```diff
@@ -0,0 +1,146 @@
+package reporter
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/go-errors/errors"
+	"github.com/smithy-security/pkg/env"
+
+	"github.com/smithy-security/smithy/sdk/component"
+	vf "github.com/smithy-security/smithy/sdk/component/vulnerability-finding"
+	ocsffindinginfo "github.com/smithy-security/smithy/sdk/gen/ocsf_ext/finding_info/v1"
+)
+
+type (
+	Conf struct {
+		SlackWebhook string
+	}
+
+	slackLogger struct {
+		Conf *Conf
+	}
+
+	SlackRequestBody struct {
+		Text string `json:"text"`
+	}
+)
+
+// NewSlackLogger returns a new slack logger.
+func NewSlackLogger(c *Conf) (*slackLogger, error) {
+	if c == nil {
+		return nil, errors.Errorf("configuration is nil")
+	}
+	return &slackLogger{
+		Conf: c,
+	}, nil
+}
+
+// NewConf returns a new configuration build from environment lookup.
+func NewConf(envLoader env.Loader) (*Conf, error) {
+	var envOpts = make([]env.ParseOption, 0)
+	if envLoader != nil {
+		envOpts = append(envOpts, env.WithLoader(envLoader))
+	}
+
+	webhook, err := env.GetOrDefault(
+		"SLACK_WEBHOOK",
+		"",
+		append(envOpts, env.WithDefaultOnError(false))...,
+	)
+	if err != nil {
+		return nil, errors.Errorf("could not get env variable for SLACK_WEBHOOK: %w", err)
+	}
+
+	return &Conf{
+		SlackWebhook: webhook,
+	}, nil
+}
+
+// Report logs the findings summary in slack.
+func (s slackLogger) Report(
+	ctx context.Context,
+	findings []*vf.VulnerabilityFinding,
+) error {
+	logger := component.
+		LoggerFromContext(ctx).
+		With(slog.Int("num_findings", len(findings)))
+
+	if len(findings) > 0 {
+		scanID := findings[0].Finding.FindingInfo.Uid
```

Comment:
How come we don't do it for all findings? I guess we could handle that with a for loop

Reply:
my assumption was that this is called for a single scan so everything should have the same scanID